addon/services/indexed-db.js

```diff
@@ -609,26 +610,36 @@ export default class IndexedDbService extends Service {
       // If a multi index is found, use it
       if (multiIndex) {
         let keyPath = get(multiIndex, 'keyPath');
-        let compareValues = array();
-        keyPath.forEach((key) => {
-          compareValues.push(query[key]);
-        });
+        let compareValues = keyPath.map((key) => query[key]);
 
         let keyName = get(multiIndex, 'name');
         return db[type].where(keyName).equals(compareValues);
       }
     }
 
     // Else, filter manually
-    Object.keys(query).forEach((i) => {
-      if (!promise) {
-        promise = db[type].where(i).equals(query[i]);
-      } else {
-        promise = promise.and((item) => get(item, i) === query[i]);
-      }
-    });
-
-    return promise;
+    return Object.entries(query)
+      .reduce(
+        (query, [queryKey, queryValue], i) => {
+          let index = indexMap[queryKey];
+
+          assert(index, `You are attempting to query ${queryKey} which is not a valid index for ${type}`);
```

Comment:
The order of arguments is wrong here - should be the message first, then the check :) so 

```js
assert(`...`, !!index)]
```

Reply:
Whoops!